--- a/llvm/lib/Support/CMakeLists.txt
+++ b/llvm/lib/Support/CMakeLists.txt
@@ -135,6 +135,8 @@
 endif()
 
 add_subdirectory(BLAKE3)
+
+list(APPEND LLVM_OPTIONAL_SOURCES raw_socket_stream.cpp)
 
 add_llvm_component_library(LLVMSupport
   ABIBreak.cpp
@@ -276,7 +278,7 @@
   YAMLTraits.cpp
   raw_os_ostream.cpp
   raw_ostream.cpp
-  raw_socket_stream.cpp
+  $<$<NOT:$<PLATFORM_ID:PlantOS>>:raw_socket_stream.cpp>
   regcomp.c
   regerror.c
   regexec.c
