--- a/src/lib.rs
+++ b/src/lib.rs
@@ -207,7 +207,7 @@
         return 0;
     }
     unsafe {
-        let terminal = Box::from_raw(terminal as *mut Terminal<Display>);
+        let terminal = &*(terminal as *const Terminal<Display>);
         terminal.rows()
     }
 }
@@ -218,7 +218,7 @@
         return 0;
     }
     unsafe {
-        let terminal = Box::from_raw(terminal as *mut Terminal<Display>);
+        let terminal = &*(terminal as *const Terminal<Display>);
         terminal.columns()
     }
 }
@@ -238,11 +238,9 @@
 }
 
 #[unsafe(no_mangle)]
-pub extern "C" fn terminal_process(terminal: *mut c_void, s: *const c_char) {
+pub extern "C" fn terminal_process(terminal: *mut c_void, s: *const u8, len: usize) {
     with_terminal!(terminal, t => {
-        if let Ok(s) = unsafe { CStr::from_ptr(s).to_str() } {
-            t.process(s.as_bytes());
-        }
+        t.process(unsafe { core::slice::from_raw_parts(s, len) });
     });
 }
 
